In [1]:
import pandas as pd

# Carregando o dataset de produtos 

df = pd.read_csv("datasets/olist_products_dataset.csv")

df


product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  
0                  14.0  
1                  20.0  
2                  15.0  
3                  26.0  
4                  13.0  
...                 ...  
32946              40.0  
32947              16.0  
32948              27.0  
32949              20.0  
32950               7.0  

[32951 rows x 9 columns]

Há dados faltantes neste dadaset?

In [2]:

df.isnull().any().any()

True

Quantos dados faltantes temos no dataset?

In [3]:
df.isnull().sum().sum()

2448

Retirar os dados faltantes

In [4]:
df = df.dropna()

Quais são as categorias de protudos 

In [5]:
categorias = df['product_category_name']
categorias



0                               perfumaria
1                                    artes
2                            esporte_lazer
3                                    bebes
4                    utilidades_domesticas
                       ...                
32946                     moveis_decoracao
32947    construcao_ferramentas_iluminacao
32948                      cama_mesa_banho
32949               informatica_acessorios
32950                      cama_mesa_banho
Name: product_category_name, Length: 32340, dtype: object

Temos 32951 dados no dataset em diversas caterigorias que podem ser as mesmas ou nao

In [6]:
#  Qual é a categoria que mais vende?
categorias.describe()


count               32340
unique                 73
top       cama_mesa_banho
freq                 3029
Name: product_category_name, dtype: object

Ou seja, existem 73 categorias úncias dentro do dataset 

In [7]:
categorias.value_counts()

product_category_name
cama_mesa_banho                  3029
esporte_lazer                    2867
moveis_decoracao                 2657
beleza_saude                     2444
utilidades_domesticas            2335
                                 ... 
fashion_roupa_infanto_juvenil       5
casa_conforto_2                     5
pc_gamer                            3
seguros_e_servicos                  2
cds_dvds_musicais                   1
Name: count, Length: 73, dtype: int64

A categoria que mais tem itens é a casa_mesa_banho, e a com menos é a de cds_dvds_musicais

Agora vamos calcular o volume dos produtos 

In [8]:
df['volume_produto'] = df['product_length_cm'] * df['product_height_cm'] * df['product_width_cm']

C:\Users\Drac3\AppData\Local\Temp\ipykernel_28452\3362806198.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['volume_produto'] = df['product_length_cm'] * df['product_height_cm'] * df['product_width_cm']


In [9]:
df

product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  volume_produto  
0                  14.0          2240.0  
1                  20.0         10800.0  
2                  15.0          2430.0  
3                  26.0          2704.0  
4                  13.0          4420.0  
...                 ...             ...  
32946              40.0         64000.0  
32947              16.0          4864.0  
32948              27.0          5103.0  
32949              20.0          8060.0  
32950               7.0           168.0  

[32340 rows x 10 columns]

E vamos descobrir os produtos com maior volume

In [10]:
#maiores produtos

df_sorted = df.sort_values(by='volume_produto', ascending=False)
df_sorted.head(50)

product_id  \
12802  256a9c364b75753b97bee410c9491ad8   
25524  c1e0531cb1864fd3a0cae57dca55ca80   
17303  3eb14e65e4208c6d94b7a32e41add538   
10697  0b48eade13cfad433122f23739a66898   
12146  f227e2d44f10f7dad30fb4dfa839e7a2   
32532  90c1b4e040d1d1c45897ec2dad4a809d   
16728  99ff40856c47a638df807c0a144470cc   
3770   c6fdec160d0f8f488d9041316c85051d   
7234   8d6f2c3454002d3f5aa7479a7fad7794   
28145  0e9dfb804bafa3d68ef3ee7a621abfb2   
27266  ab495f166205a883ffe5ab0b5b55f867   
20574  a3cd9517ebf5a50dca25acce54f3b171   
23988  ab50aa8d9af0addb7f96149c8b8cee86   
1159   f97ad9066c718a6cef93dfcf253d3e0d   
25978  f18f40e6344e72cfd537f21d9745e8cc   
24685  c8d4b2652543be14a917376a17f426c2   
30958  04abe952a7292c9914a0ddade6371114   
9356   671be61e4013fc77e2d3586ffc1e51f1   
17314  2a511952313989fa097fb038cd758dc5   
23942  f3f12eabcd090650146791b249f48880   
6212   e0c47a145f4748e4d7a418502d65aa35   
13987  d13816a79f03bf978283d80fbeae06f8   
6012   d5536c1422144c4fc6fe001224b21ba8   
29085  107fde0930956120d1e13dd1062fbb46   
30518  013b5dd7d319685753e2c12292975615   
1514   dcfeedf441c38e5e7e58ffce194af2bb   
217    6e9c52b407fea24064f7ab2b144fbf8c   
26970  b17808303e15dd50538c011b44295427   
10248  3d77287739b6bf1ac163ce2d77570ada   
6728   5caaa02918381c4ad59d38fdcd4a28b9   
15581  b7a64f15c532c9eacbaa11cbef1bc969   
13324  27761f21b03a32be1aad6ebaebddd747   
24886  5dcd17b12ed2175d4fe6157e047d7be2   
10463  0d5837a2580f92e18e2d1b5f7fa70c2d   
6004   368645a47a15770520ad12d9ea77c2f2   
10708  b6e406be1aa00db8be5dd3ec6b524d4d   
7535   9dfef86fb34051388a7263a31642386c   
8218   9fa7e05ae5cddeb74ff2fe5046b742db   
2537   0ed9a84687fad2d921b09b0ebcc4cded   
21443  a9b7d6445491f7108ce079d4f783dbf1   
7422   bbc5419bc92d76e27795c9ec2c233c64   
20222  48c9b5b0aff66a99d74fbc6e2c600b33   
12376  773d43a287d64831b05013a73a33f229   
4814   0984eaa8480e41aded35bd7b5131a1c1   
26495  d15010eecc80e3a843ccb15865c328da   
21484  67f1a036affe9a75b73dfddf0386893a   
27326  46e24ce614899e36617e37ea1e4aa6ff   
24029  1d16c0509e7011656e03c6a582ff6667   
27698  97c948ebc8c04b26b7bbb095d4228f2a   
29578  dc861f18267cce21f3b70f9f7990b8ce   

                                product_category_name  product_name_lenght  \
12802                           utilidades_domesticas                 59.0   
25524                                     moveis_sala                 50.0   
17303                                     moveis_sala                 51.0   
10697                                moveis_decoracao                 28.0   
12146                                     moveis_sala                 51.0   
32532                                     moveis_sala                 57.0   
16728                                     moveis_sala                 54.0   
3770                                      moveis_sala                 57.0   
7234                                      moveis_sala                 52.0   
28145                                     moveis_sala                 17.0   
27266                                eletrodomesticos                 57.0   
20574                           utilidades_domesticas                 52.0   
23988                                      cool_stuff                 34.0   
1159                                 moveis_decoracao                 63.0   
25978                                moveis_decoracao                 63.0   
24685                                  consoles_games                 41.0   
30958                                        pet_shop                 33.0   
9356                                    esporte_lazer                 28.0   
17314                                    beleza_saude                 51.0   
23942                                   esporte_lazer                 47.0   
6212                                       cool_stuff                 56.0   
13987                                     moveis_sala                 45.0   
6012                                 move

Os produtos de maior volume são os invlusos como beleza_saude e moveis_saa

In [11]:
df['volume_produto'].sort_values(ascending=False)

12802    296208.0
25524    294000.0
17303    294000.0
10697    294000.0
12146    294000.0
           ...   
4415        352.0
20028       352.0
20041       352.0
961         288.0
32950       168.0
Name: volume_produto, Length: 32340, dtype: float64

o volume maior é de 296208.0 cm3
e o de menor é de 168cm3

Agora vamos calcular os produtos mais pesados 

In [12]:
#produtos mais pesados

df_sorted = df.sort_values(by='product_weight_g', ascending=False)
df_sorted.head(50)


product_id  \
25166  26644690fde745fc4654719c3904e1db   
24268  cb26d15d1b6eabaac7c0803774245884   
1701   1c57458e824ca3d974ec1831a1a55e72   
27266  ab495f166205a883ffe5ab0b5b55f867   
19619  cdb8d3c880b6639a70764c6734e6bb69   
1714   c04e948c6900ce99ac47d89b3b6d70cd   
9363   8f73613d06e3a557da0249015cbae6b6   
12376  773d43a287d64831b05013a73a33f229   
32262  ebf1c13032246ea801765e8cb5417365   
27230  e5f2d52b802189ee658865ca93d83a8f   
9373   038e3a669bb495e56d9d609d874fb668   
1738   0a859d8dc68f6a746b4709217110c439   
19753  31d22f91c17551712034a105e70ab838   
11327  8d4e92265a16e69a1e1d76e67e46d72f   
7535   9dfef86fb34051388a7263a31642386c   
19834  0f3777229e40207d248acfd6e1f716b6   
8218   9fa7e05ae5cddeb74ff2fe5046b742db   
8644   59f09deeea7c839a9212a19f3416b892   
19875  9f82fdef2a36bb15d22d70af1aedb3ce   
32179  d7f8b7443f6ae5796dabe493521a1e79   
5465   df872c596e00cd0160e6c8eb5424739f   
26912  bad9cd5ad615c0b5ba87448e03ec954c   
30340  660422061e06da17ca6101e9d6b7aae8   
14992  07f7c5fe95aa4a3b8ea56a5119546939   
20222  48c9b5b0aff66a99d74fbc6e2c600b33   
9356   671be61e4013fc77e2d3586ffc1e51f1   
2417   343c15a347e523f2b6cf38a5db81e179   
1514   dcfeedf441c38e5e7e58ffce194af2bb   
27326  46e24ce614899e36617e37ea1e4aa6ff   
27858  0c5aecba8eab00eb055c6e2e7b9b4346   
16575  ab14f3763110d218ea7862647bd450af   
32532  90c1b4e040d1d1c45897ec2dad4a809d   
18791  fe5915b903037937488e868f2dde0ff3   
27698  97c948ebc8c04b26b7bbb095d4228f2a   
32491  2b752ed328ea866e4721ca4e236a416c   
29578  dc861f18267cce21f3b70f9f7990b8ce   
24861  961930544d39899c18cfdbc336aeb15e   
16450  e6506b15fb88b22adb8f9dcdb20e532a   
23581  a8baceb529f7e2a5c770cc5b4e3da35d   
16365  f18e0f3757837733f36da5143ddccd73   
12105  c75c215de0f9ab2c103fc57fb82f47ad   
29682  17383e180cbea9ef103d90a72754cd54   
19023  bc75ffbc500b02276f9c55aef39014e9   
12146  f227e2d44f10f7dad30fb4dfa839e7a2   
16206  df4566130f96bba07730095eef42389e   
7422   bbc5419bc92d76e27795c9ec2c233c64   
19233  1ab21d1c80e214bc2cedfcb7a5ade550   
27449  7d96d1abbde52bb350c688d0d049df07   
4185   c3eb7064c36408a9bfc77a42d6ab993c   
15982  c100e5fef1abb5e1c5054d1dac2d83ac   

                                product_category_name  product_name_lenght  \
25166                                 cama_mesa_banho                 59.0   
24268                                   esporte_lazer                 54.0   
1701                                         pet_shop                 57.0   
27266                                eletrodomesticos                 57.0   
19619                                    beleza_saude                 25.0   
1714                                       cool_stuff                 33.0   
9363                                moveis_escritorio                 54.0   
12376                                    beleza_saude                 55.0   
32262                                 casa_construcao                 39.0   
27230                                        pet_shop                 56.0   
9373                                    esporte_lazer                 49.0   
1738                                    esporte_lazer                 50.0   
19753                                moveis_decoracao                 29.0   
11327                           instrumentos_musicais                 45.0   
7535                                     beleza_saude                 57.0   
19834                                   esporte_lazer                 64.0   
8218                                     beleza_saude                 55.0   
8644                            utilidades_domesticas                 58.0   
19875                              ferramentas_jardim                 57.0   
32179                   industria_comercio_e_negocios                 56.0   
5465   moveis_cozinha_area_de_servico_jantar_e_jardim                 39.0   
26912                                    beleza_saude                 59.0   
30340                                   e

A maioria dos produtos pesados são o esporte_lazer e beleza_saude	

In [13]:
df['product_weight_g'].sort_values(ascending=False)

25166    40425.0
24268    30000.0
1701     30000.0
27266    30000.0
19619    30000.0
          ...   
15675        2.0
14997        0.0
9769         0.0
32079        0.0
13683        0.0
Name: product_weight_g, Length: 32340, dtype: float64

In [18]:
df['product_weight_g'].describe()

count    32340.000000
mean      2276.956586
std       4279.291845
min          0.000000
25%        300.000000
50%        700.000000
75%       1900.000000
max      40425.000000
Name: product_weight_g, dtype: float64

O produto mais pesado é de 40425 gramas e de menor 0 gramas 

Adicionar uma nova coluna de acordo com o peso
40425 - 26950 - Pesado
26950 - 13475 - Peso mediano 
13475 - 0 - Leve 

In [25]:

def categoriando_peso(x):
    if x > 26950:
        return 'Pesado'
    elif x >= 13475 and x <= 26950:
        return 'Peso mediado'
    else:
        return 'Leve'


df['categorizar_peso'] = df['product_weight_g'].apply(categoriando_peso)

C:\Users\Drac3\AppData\Local\Temp\ipykernel_28452\1499920177.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['categorizar_peso'] = df['product_weight_g'].apply(categoriando_peso)


In [26]:
df.head(50)

product_id             product_category_name  \
0   1e9e8ef04dbcff4541ed26657ea517e5                        perfumaria   
1   3aa071139cb16b67ca9e5dea641aaa2f                             artes   
2   96bd76ec8810374ed1b65e291975717f                     esporte_lazer   
3   cef67bcfe19066a932b7673e239eb23d                             bebes   
4   9dc1a7de274444849c219cff195d0b71             utilidades_domesticas   
5   41d3672d4792049fa1779bb35283ed13             instrumentos_musicais   
6   732bd381ad09e530fe0a5f457d81becb                        cool_stuff   
7   2548af3e6e77a690cf3eb6368e9ab61e                  moveis_decoracao   
8   37cc742be07708b53a98702e77a21a02                  eletrodomesticos   
9   8c92109888e8cdf9d66dc7e463025574                        brinquedos   
10  14aa47b7fe5c25522b47b4b29c98dcb9                   cama_mesa_banho   
11  03b63c5fc16691530586ae020c345514                             bebes   
12  cf55509ea8edaaac1d28fdb16e48fc22             instrumentos_musicais   
13  7bb6f29c2be57716194f96496660c7c2                  moveis_decoracao   
14  eb31436580a610f202c859463d8c7415  construcao_ferramentas_seguranca   
15  3bb7f144022e6732727d8d838a7b13b3                     esporte_lazer   
16  6a2fb4dd53d2cdb88e0432f1284a004c                        perfumaria   
17  a1b71017a84f92fd8da4aeefba108a24            informatica_acessorios   
18  a0736b92e52f6cead290e30b578413b6                  moveis_decoracao   
19  f53103a77d9cf245e579ea37e5ec51f0                   cama_mesa_banho   
20  1c1890ba1779090cd54008a3c3302921                  moveis_decoracao   
21  518ef5de2c2b3a255e326a4594ba15d9                   cama_mesa_banho   
22  e3e020af31d4d89d2602272b315c3f6e                      beleza_saude   
23  8b41fbc2b984a12030090112324d1bc4                             bebes   
24  d03bd02af9fff4b98f1c972315e5e9ef                  moveis_decoracao   
25  8ba4f2a4ae695d26e5626c1bf710975e             utilidades_domesticas   
26  f6574524b07d5bb753bab6f35abe2d06                  malas_acessorios   
27  c78b767da00efb70c1bcccab87c28cd5            informatica_acessorios   
28  a0253d43394dd4da9a5d7b1f546f1a32            informatica_acessorios   
29  051b9ff13dd55c0a6655a15ff296f80d                     esporte_lazer   
30  47859fca9dd7485cbd93c3e8993bb74f                   cama_mesa_banho   
31  e1d1d22e9f8122a4ec1533b032c12562                ferramentas_jardim   
32  20e37962088a4061d3578def47a8648b                  moveis_decoracao   
33  1eba879220bd0981a0e2fbab499ed4e1                 moveis_escritorio   
34  ce5b91848b91118daffb3af53b747475                     esporte_lazer   
35  5370b82a213393979691c98074265584                             bebes   
36  b8dac5113b06a97e64943234522572b9                        automotivo   
37  750cf819d127191920eda79a4b6fb479                       eletronicos   
38  e6af694343b45b56304ad91974a110b9             utilidades_domesticas   
39  680874c570dad71c0a2844cfbf417054                  moveis_decoracao   
40  8b3a9476f74f5297f7ff0ec6d95fe1ea                  moveis_decoracao   
41  1c6fb703c624b381a20f21f757694866                        brinquedos   
42  db46052aed7dff59ba38cf2fe29b87f9                   cama_mesa_banho   
43  b864b103180d5a6a3f830e67f93e8735                             bebes   
44  7a8dac4aaa16bc642e4df33adcf03303                        cool_stuff   
45  83b9bc6aae6f527ff6aafb9e01d6cbf3                        brinquedos   
46  5f504b3a1c75b73d6151be81eb05bdc9                        cool_stuff   
47  bbf920aa6ac72007a726e6f419156f8f                  moveis_decoracao   
48  c3b7e4d83e530c27a8e6368296aece48                  moveis_decoracao   
49  c5d8079278e912d7e3b6beb48ecb56e8                      beleza_saude   

    product_name_lenght  product_description_lenght  product_photos_qty  \
0                  40.0                       287.0                 1.0   
1                  44.0                       276.0                 1.0   
2                  46.0   

Proporção dos pesos entre as categorias leve, peso mediano e pesado

In [27]:
df['categorizar_peso'].value_counts()

categorizar_peso
Leve            31166
Peso mediado      974
Pesado            200
Name: count, dtype: int64

Aqui fizemos uma análise de quantos caracteres são usados geralmente para um produtos


In [14]:
df['product_name_lenght'].describe()

count    32340.000000
mean        48.476592
std         10.245699
min          5.000000
25%         42.000000
50%         51.000000
75%         57.000000
max         76.000000
Name: product_name_lenght, dtype: float64

A média é de 48 caracteres

Fizemos a mesma coisa para a descrição dos protudos

In [ ]:
df['product_description_lenght'].describe()

A média é de 771 caracteres

Aqui estamos buscando o produto que tem a maior descrição

In [ ]:
max_product_description_lenght = df['product_description_lenght'].max()
df_product_description_lenght = df.loc[df['product_description_lenght'] == max_product_description_lenght]
df_product_description_lenght

Vamos encontrar quantas fotos geralmente são fornecidadas

In [ ]:
df['product_photos_qty'].describe()

A média é geralmente de 2 fotos para cada produto

Qual é a o produto que tem mais fotos?

In [ ]:
max_photos_qty = df['product_photos_qty'].max()
df_max_photos = df.loc[df['product_photos_qty'] == max_photos_qty]
df_max_photos

Aqui vamos ler um novo dataser do nosso bando de dados

In [ ]:
import pandas as pd

df_items = pd.read_csv("datasets/olist_order_items_dataset.csv")

df_items

Remover os dados vazios se existirem

In [ ]:
df.isnull().any().any()

Neste caso não existem dados vazios ou nulos

Vamos unir as duas tabelas, que será chamada de df_merged

In [ ]:
df_merged = pd.merge(df, df_items, on='product_id', how='inner')
df_merged

In [ ]:
df_merged.describe()

In [ ]:
df_merged.value_counts()

Aqui vamos ver dados sobre o preço dos produtos

In [ ]:
df_merged['price'].describe()

In [ ]:
# quais os produtos mais caros?

df_merged_sorted_price_max = df_merged.sort_values(by='price', ascending=False)
df_merged_sorted_price_max.head(50)


In [ ]:
# quais os produtos mais baratos?

df_merged_sorted_price_min = df_merged.sort_values(by='price', ascending=True)
df_merged_sorted_price_min.head(50)


Aqui vamos ver dados sobre o preço do frete dos produtos

In [ ]:
df_merged['freight_value'].describe()

In [ ]:
# quais os produtos com maior preço de frete?

df_merged_sorted_frete_max = df_merged.sort_values(by='freight_value', ascending=False)
df_merged_sorted_frete_max.head(50)

In [ ]:
# quais os produtos com menor preço de frete?

df_merged_sorted_frete_min = df_merged.sort_values(by='freight_value', ascending=True)
df_merged_sorted_frete_min.head(50)


Calculando os produtos que são mais caros que a média do preço

In [ ]:

media_preco = df_merged['price'].mean()
print(f"A média do preço dos produtos é: {media_preco:.2f}")
quant_preco = (df_merged['price'] > media_preco).sum()
print(f"A quantidade de produtos maior que a é a média do preço: {quant_preco}")


Calculando os produtos que o frete é mais caro que a média

In [ ]:
media_preco_frete = df_merged['freight_value'].mean()
print(f"A média do preço do frete é: {media_preco_frete:.2f}")
quant_preco_frete = (df_merged['freight_value'] > media_preco_frete).sum()
print(f"A quantidade de produtos maior que a é a média do preço: {quant_preco_frete}")


Aqui vamos plotar um gráfico de pontos para ver se existe uma relação entre o volume e o frete

In [ ]:
# volume x frete 

import matplotlib.pyplot as plt


plt.figure(figsize=(10, 6)) 
plt.scatter(df_merged['volume_produto'], df_merged['freight_value']) 


plt.title('Gráfico de Dispersão entre X e Y')
plt.xlabel('X')
plt.ylabel('Y')


plt.show()

Olhando o gráfico parece não existir uma correspondencia entre o volume do produto e o valor do frete

In [ ]:
# peso x frete 



plt.figure(figsize=(10, 6))
plt.scatter(df_merged['product_weight_g'], df_merged['freight_value']) 


plt.title('Gráfico de Dispersão entre X e Y')
plt.xlabel('X')
plt.ylabel('Y')


plt.show()

Olhando o gráfico também parece não existir uma correspondencia entre o peso do produto e o valor do frete